### ISSUE TITLE
Not able to delete an aggregate using Aggregates API - Sensu 0.20.3

### ISSUE THREAD

**Issue by p1hr** [sensu/sensu#1070]
Hi,
I'm using Sensu 0.20.3 version. 

I currently have a list of aggregates on my server as below. Using Aggregates API I am not able to delete aggregates that are upper case like JAVA, APACHE etc - the API returns 404 Not Found. The same DELETE request works ok (returning 204) on lower case aggregates.
```
[
  {
    "check": "check_mounts",
    "issued": []
  },
  {
    "check": "cron",
    "issued": []
  },
  {
    "check": "apache",
    "issued": []
  },
  {
    "check": "check_cron",
    "issued": []
  },
  {
    "check": "check_ssh",
    "issued": []
  },
  {
    "check": "JAVA",
    "issued": []
  },
  {
    "check": "mounts",
    "issued": []
  },
  {
    "check": "APACHE",
    "issued": []
  },
  {
    "check": "java",
    "issued": []
  },
  {
    "check": "MOUNTS",
    "issued": []
  },
  {
    "check": "tomcat",
    "issued": []
  },
  {
    "check": "updates",
    "issued": []
  },
  {
    "check": "SSH",
    "issued": []
  },
  {
    "check": "check_java",
    "issued": []
  },
  {
    "check": "CRON",
    "issued": []
  },
  {
    "check": "check_apache",
    "issued": []
  },
  {
    "check": "ssh",
    "issued": []
  },
  {
    "check": "check_updates",
    "issued": []
  },
  {
    "check": "UPDATES",
    "issued": []
  }
]
````

**Comment by portertech** [comment]
@JackBauer8301 Thank you for reporting this bug :+1: 

Perhaps the endpoint regex - https://github.com/sensu/sensu/blob/master/lib/sensu/api/process.rb#L607

**Comment by JackBauer8301** [comment]
hi @portertech ,
that was my initial idea so I played with the regex a bit so that it matches both lower-case and upper-case characters but I couldn't make it work

Things I've tried:
```
%r{^(i?)/aggregates/([\w\.-]+)/?$} do |check_name| 
%r{^/aggregates/([\A-Za-z\.-]+)/?$} do |check_name| 
```

What seems weird to me is that I manually deleted all keys related to those aggregates from Redis but they still appear in Uchiwa.

**Comment by portertech** [comment]
Let's add a Sensu API spec for this and see if it is due to the API itself and guard against regressions. :+1:

**Comment by calebhailey** [comment]
Tagging for 0.24.

**Comment by portertech** [comment]
https://github.com/sensu/sensu/pull/1301

**Comment by portertech** [comment]
https://github.com/sensu/sensu/pull/1301 has been merged, tests pass with the 0.24 Aggregates 2.0 implementation :+1:
